src/submitter.py

```diff
@@ -423,23 +441,36 @@ def handler(event, context):
             continue
 
         LOGGER.debug("saw crash id: %s in %s", crash_id, bucket)
-
-        # Throttle crashes
-        if CONFIG.throttle < 100 and random.randint(0, 100) > CONFIG.throttle:
-            LOGGER.info("submitted: %s", crash_id)
-            statsd_incr("socorro.submitter.throttled", value=1)
-            continue
-
-        accepted_records.append(crash_id)
+        crash_ids.append(crash_id)
 
     # If we don't have anything to post, we're done!
-    if not accepted_records:
+    if not crash_ids:
         return
 
-    for crash_id in accepted_records:
-        try:
+    destinations = CONFIG.get_destinations()
+
+    for crash_id in crash_ids:
+        submit_destinations = []
+
+        # Figure out which destinations we're sending to
+        for destination in destinations:
+            if (
+                destination.throttle < 100
+                and random.randint(0, 100) > destination.throttle
+            ):
+                LOGGER.info("throttled: %s (%r)", crash_id, destination)
+                statsd_incr("socorro.submitter.throttled", value=1)
+                continue
+
             statsd_incr("socorro.submitter.accept", value=1)
+            submit_destinations.append(destination)
+
+        # If there's nowhere to send to, move on to next crash_id
+        if not submit_destinations:
+            continue
 
+        # Fetch the crash report data
+        try:
             # Build client
             client = build_s3_client(
```

Comment:
should we be building the client outside this loop instead of for every client id?

Reply:
Oops--I missed this one. I'll fix this.